sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java

```diff
@@ -0,0 +1,197 @@
+    super(data.file);
+    currentSpan = data.span;
+    file = data.file;
+    delegate = data.delegate;
+  }
+
+  private static FileInputStreamInitData init(
+    @Nullable File file,
+    @Nullable FileInputStream delegate
+  ) throws FileNotFoundException {
+    final ISpan span = startSpan();
+    if (delegate == null) {
+      delegate = new FileInputStream(file);
+    }
+    return new FileInputStreamInitData(file, span, delegate);
+  }
+
+  private static FileInputStreamInitData init(
+    @NotNull FileDescriptor fd,
+    @Nullable FileInputStream delegate
+  ) {
+    final ISpan span = startSpan();
+    if (delegate == null) {
+      delegate = new FileInputStream(fd);
+    }
+    // TODO: it's only possible to get filename from FileDescriptor via reflection AND when it's
+    // TODO: running on Android, should we do that?
+    return new FileInputStreamInitData(null, span, delegate);
+  }
+
+  private static @Nullable
+  ISpan startSpan() {
+    final ISpan parent = Sentry.getSpan();
+    return parent != null ? parent.startChild("file.read") : null;
+  }
+
+  @Override
+  public int read() throws IOException {
+    try {
+      int result = delegate.read();
+      if (result != -1) {
+        byteCount++;
+      }
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public int read(byte @NotNull [] b) throws IOException {
+    try {
+      int result = delegate.read(b);
+      if (result != -1) {
+        byteCount += result;
+      }
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public int read(byte @NotNull [] b, int off, int len) throws IOException {
+    try {
+      int result = delegate.read(b, off, len);
+      if (result != -1) {
+        byteCount += result;
+      }
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public long skip(long n) throws IOException {
+    try {
+      long result = delegate.skip(n);
+      byteCount += result;
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    try {
+      delegate.close();
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+    finishSpan();
+  }
+
+  private void finishSpan() {
+    if (currentSpan != null) {
+      if (file != null) {
+        String description = file.getName()
+          + " "
+          + "("
+          + StringUtils.byteCountToString(byteCount)
+          + ")";
+        currentSpan.setDescription(description);
+        currentSpan.setData("filepath", file.getAbsolutePath());
```

Comment:
wondering if it makes sense to be relative path because of PII

Reply:
Good point, I think for Android it's not concerning, since it's always under `data/user/0/etc.`, but for pure JVM/backend projects can be dangerous. @bruno-garcia opinions?

Reply:
I think it's valuable to have the FQP. The problem is knowing where in the path we should cut off not to throw away valuable info. There are some heuristics like `/Users/Xyz` and `C:\Users\Xyz` and IIRC that was done on the server.

There's PII filtering in ingest but I'm not sure if it'll apply to Spans. @untitaker  might be able to help us here.

Reply:
Hi, I replied in Slack because I am not sure of the answer, but this is what I am sure of:

> 
> If you can't find a reasonable cutoff in platform- and domain-specific code, the server will do no better. The [advanced data scrubbing rules](https://docs.sentry.io/product/data-management-settings/scrubbing/advanced-datascrubbing/) you can configure at the moment are, and yes they will also work on span data:
> 
> 1. set a custom regex to strip away matches
> 2. strip the entire path and leave the basename
> 
> This has to be explicitly set up by the user. However, we communicate in all SDK docs that our SDKs do not send PII by default. So you cannot intentionally send PII and just expect the server to strip it away. That's not the guarantee we make, and even if the server does manage to strip away all the PII it will be too late for some customers. Safe defaults first.

That doesn't necessarily resolve your discussion because it's unclear to me at the moment as to whether we consider absolute paths as "PII".

Reply:
I'd rather go with file name then, better safe than sorry.
@romtsn maybe confirm with Jan/Armin, they might know if the file path is PII or not

Reply:
looks like we can easily distinguish between desktop vs server using https://docs.oracle.com/javase/8/docs/api/java/awt/GraphicsEnvironment.html#isHeadless--, so maybe we just skip absolute paths for desktops

Reply:
this wont compile for Android, awt isn't part of the Android SDK iirc, unless you add a layer of abstraction.
check `io.sentry.util.Platform#isAndroid` tho, it'd be fine to add for Mobile I believe at least.

Reply:
good concern. There was also mentioned `sendDefaultPii` in the slack thread, so perhaps, we could send it for server and desktop, if this flag is `true`, and always for mobile